src/main/java/seedu/edrecord/logic/parser/AddAssignmentCommandParser.java

```diff
@@ -0,0 +1,49 @@
+package seedu.edrecord.logic.parser;
+
+import static seedu.edrecord.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.edrecord.logic.parser.CliSyntax.PREFIX_MAX_SCORE;
+import static seedu.edrecord.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.edrecord.logic.parser.CliSyntax.PREFIX_WEIGHTAGE;
+
+import java.util.stream.Stream;
+
+import seedu.edrecord.logic.commands.AddAssignmentCommand;
+import seedu.edrecord.logic.parser.exceptions.ParseException;
+import seedu.edrecord.model.assignment.Assignment;
+import seedu.edrecord.model.assignment.MaxScore;
+import seedu.edrecord.model.assignment.Weightage;
+import seedu.edrecord.model.name.Name;
+
+public class AddAssignmentCommandParser implements Parser<AddAssignmentCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the {@code MakeModuleCommand}
+     * and returns a {@code MakeModuleCommand} object for execution.
```

Comment:
Should it be `AddAssignmentCommand` in the header comment?
```suggestion
     * Parses the given {@code String} of arguments in the context of the {@code AddAssignmentCommand}
     * and returns a {@code AddAssignmentCommand} object for execution.
```